src/Examine/LuceneEngine/WriterTracker.cs

```diff
@@ -41,9 +44,71 @@ public IndexWriter GetWriter(Directory dir, bool throwIfEmpty)
 
         public IndexWriter GetWriter(Directory dir, Func<Directory, IndexWriter> factory)
         {
-            var resolved = _writers.GetOrAdd(dir.GetLockId(), s => factory(dir));
-            return resolved;
+            lock (_locker)
+            {
+                var resolved = _writers.GetOrAdd(dir.GetLockId(), s => factory(dir));
+                return resolved;
+            }
         }
 
+        public void MapAsOld(Directory dir)
```

Comment:
Some code docs explaining what this does would be good.

Reply:
@Shazwazza will add :)

Reply:
Commented added, please let know if you need more descriptive comment